docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,468 @@
+
+> **Good to know**: Check if your upstream data provider has caching enabled by default. You might need to disable (e.g. `useCdn: false`), otherwise a revalidation won't be able to pull fresh data to update the Data Cache. Caching can occur at a CDN (for an endpoint being requested) when it returns the `Cache-Control` header. ISR on Vercel [persists the cache globally and handles rollbacks](https://vercel.com/docs/concepts/incremental-static-regeneration/overview).
+
+#### Opting out
+
+For individual data fetches, you can opt out of caching by setting the [`cache`](#fetchoptionscache) option to `no-store`. This means data will be fetched dynamically, on each request.
+
+```jsx
+// Opt out of caching
+fetch(`https://...`, { cache: 'no-store' })
+```
+
+Alternatively, you can also use the [Route Segment Config options](#segment-config-options) to opt out of caching for **all** data requests in a route segment.
+
+> **Vercel Data Cache**
+>
+> If your Next.js application is deployed to Vercel, we recommend reading the [Vercel Data Cache](https://vercel.com/docs/infrastructure/data-cache) documentation for a better understanding of Vercel specific features.
+
+### Full Route Cache
+
+Next.js automatically caches the rendered result of React Components on the server at build time. This is an optimization that allows you to skip rendering components on the server on every request, and instead directly send the cached result to the client, resulting in faster page loads.
+
+> **Related terms**:
+>
+> - The terms **Automatic Static Optimization**, **Static Site Generation**, or **Static Rendering** might be also be used interchangeably to refer to the process of rendering and caching parts of your application code on the server at build time.
+
+To understand how the Full Route Cache works, it's helpful to look at how React handles rendering, and how Next.js caches the result:
+
+#### 1. React Rendering on the Server
+
+On the server, Next.js uses React's APIs to orchestrate rendering. This process is split into three stages:
+
+1. React renders Server Components into a special data format, optimized for streaming, called the **React Server Component Payload**.
+2. Next.js compiles, bundles, and code-splits the **JavaScript instructions** for rendering Client Components on the client.
+3. Next.js uses the React Server Component Payload and JavaScript instructions to pre-render **HTML** on the server.
+
+> **What is React Server Component Payload?**
+>
+> The React Server Component Payload is a compact string representation of the React Component tree. It's used by React on the client to update the DOM. The React Server Component Payload contains:
+>
+> - The rendered result of Server Components
+> - Placeholders for where Client Components should be rendered on the client
```

Comment:
```suggestion
> - Placeholders for where Client Components should be rendered
```

Let's remove "on the client" as it can fuel confusion around these being rendered to HTML.